password_enhancements.module

```diff
@@ -522,8 +553,277 @@ function password_enhancements_legal_allowed_paths_alter(array &$allowed_paths):
+      '#states' => [
+        'invisible' => [
+          ':input[name="password_enhancements[expire_warn_before_days]"]' => ['value' => 0],
+        ],
+      ],
+    ];
+
+    // Only allow adding constraints when the policy is already set up.
+    if ($policy) {
+      // Build the list of existing password constraints for this password
+      // policy.
+      $form['password_enhancements']['constraints'] = [
+        '#type' => 'table',
+        '#header' => [
+          t('Constraint'),
+          t('Required'),
+          t('Summary'),
+          t('Operations'),
+        ],
+        '#attributes' => [
+          'id' => 'password-policy-constraints',
+        ],
+        '#empty' => t('There are currently no constraints in this policy. Add one by selecting an option below.'),
+        // Render constraints below parent elements.
+        '#weight' => 5,
+      ];
+      $old_unique_constraints = [];
+      /** @var \Drupal\password_enhancements\PasswordConstraintInterface $constraint */
+      foreach ($policy->getConstraints() as $constraint) {
+        if ($constraint->getPluginDefinition()['unique']) {
+          $old_unique_constraints[$constraint->getPluginId()] = $constraint->getPluginId();
+        }
+        $key = $constraint->getUuid();
+        $form['password_enhancements']['constraints'][$key]['constraint'] = [
+          '#tree' => FALSE,
+          'data' => [
+            'name' => [
+              '#plain_text' => $constraint->name(),
+            ],
+          ],
+        ];
+
+        $form['password_enhancements']['constraints'][$key]['required'] = [
+          '#plain_text' => $constraint->isRequired() ? t('Yes') : t('No'),
+        ];
+
+        if ($summary = $constraint->getSummary()) {
+          $form['password_enhancements']['constraints'][$key]['summary'] = $summary;
+        }
+
+        $links = [
+          'edit' => [
+            'title' => t('Edit'),
+            'url' => Url::fromRoute('password_enhancements.constraint_edit_form', [
+              'user_role' => $policy->getRole()->id(),
+              'password_constraint' => $key,
+            ]),
+          ],
+          'delete' => [
+            'title' => t('Delete'),
+            'url' => Url::fromRoute('password_enhancements.constraint_delete_form', [
+              'user_role' => $policy->getRole()->id(),
+              'password_constraint' => $key,
+            ]),
+          ],
+        ];
+        $form['password_enhancements']['constraints'][$key]['operations'] = [
+          '#type' => 'operations',
+          '#links' => $links,
+        ];
+      }
+
+      // Build the new password constraint addition form and add it to the
+      // constraint list.
+      $password_constraint_manager = \Drupal::service('plugin.manager.password_enhancements.constraint');
+      $new_constraint_options = [];
+      $constraint_definitions = $password_constraint_manager->getDefinitions();
```

Comment:
I don't think so, because there's no point in sorting the _definitions_ of the plugins. OTOH, the only reason for sorting them _here_ is that we want to have a consistent UI.